src/Microsoft.Azure.WebJobs.ServiceBus/EventHubs/EventHubConfiguration.cs

```diff
@@ -256,6 +270,90 @@ EventProcessorHost IEventHubProvider.GetEventProcessorHost(string eventHubName,
             throw new InvalidOperationException("No event hub receiver named " + eventHubName);
         }
 
+        private static string EscapeStorageCharacter(char character)
+        {
+            var ordinalValue = (ushort)character;
+            if (ordinalValue < 0x100)
+            {
+                return string.Format(CultureInfo.InvariantCulture, ":{0:X2}", ordinalValue);
+            }
+            else
+            {
+                return string.Format(CultureInfo.InvariantCulture, "::{0:X4}", ordinalValue);
+            }
+        }
+                
+        // Escape a blob path.  
+        // For diagnostics, we want human-readble strings that resemble the input. 
+        // Inputs are most commonly alphanumeric with a fex extra chars (dash, underscore, dot). 
+        // Escape character is a ':', which is also escaped. 
+        // Blob names are case sensitive; whereas input is case insensitive, so normalize to lower.  
+        private static string EscapeBlobPath(string path)
```

Comment:
Need to add tests for all these new helper methods. That would also help us better visualize the transformations.